### ISSUE TITLE
[Feature Request]: Julia support

### ISSUE THREAD

**Issue by gustaphe** [puremourning/vimspector#256]
Julia has the [Debugger.jl](https://github.com/JuliaDebug/Debugger.jl) package which I believe to implement debugging in a standardized way. It would be really nice to get it into vim.

I'm looking into how the entire debugger echosystem works, but I don't have a lot of time to work on it. If someone else has a working configuration I'd be happy to see it.

Anyway, now it's tracked.

**Comment by gustaphe** [comment]
[DebugAdapter.jl](https://github.com/julia-vscode/DebugAdapter.jl) seems more relevant, looking at it. That's an actual DAP implementation.

**Comment by puremourning** [comment]
https://github.com/puremourning/vimspector/wiki/languages

**Comment by gustaphe** [comment]
I've tried to look into it, but I'm really not versed in VSCode. Looks like everything tagged as "command" in `package.json` is on the form `language-julia.someCommand`, and I don't know which language that's supposed to be.

**Comment by puremourning** [comment]
In the complete absence of documentation, you have to revert to reverse engineering.

The usual process I follow is open `package.json` for the vscode extension, then search for `"debuggers"`: https://github.com/julia-vscode/julia-vscode/blob/master/package.json#L700

Usually at this point you will be told 2 things:

1. How to launch the debug adapter,
2. What launch config looks like

But in this case, there's nothing there saying how to launch it so you go down the rabbit hole and try and read the vscode extension source code:

https://github.com/julia-vscode/julia-vscode/tree/master/src/debugger

You can see that there is a typescript implementation of the adapter here: https://github.com/julia-vscode/julia-vscode/blob/master/src/debugger/juliaDebug.ts

This seems to launch the real debugger in a terminal, but _directly_ in vscode, rather than the runInTerminal request, which is annoying: https://github.com/julia-vscode/julia-vscode/blob/master/src/debugger/juliaDebug.ts#L225-L242

You'll also find that it's using some vscode-only mechanisms for running the debug adapter in-process (or something): https://github.com/julia-vscode/julia-vscode/blob/36aaed6bb7eff8431a0ffbc73645ea23646fcb80/src/debugger/debugFeature.ts#L109

Be wary of any extension which does this - it means there's almost no chance that it will work with generic DAP.

And if you see this: https://github.com/julia-vscode/julia-vscode/blob/master/src/debugger/juliaDebug.ts#L5 then there's no chance.

The DebugAdapter.jl seems to have no documentation at all, so you'll probably have to guess if that would work..

In summary:

* The vscode-julia extension contains a debug adapter implementation which only works in Vscode, despite: https://github.com/julia-vscode/julia-vscode/issues/125
* The DebugAdapter.jl doesn't appear to be finished or documented.

I suggest you raise an issue on their repo asking them to make the typescript debug adapter run outside of vscode (this looks easy - just just the DAP runInTerminal request instead of invoking vscode API directly)

**Comment by puremourning** [comment]
Closing as there does not appear to be a viable Julia debug adapter.
